packages/lambda/src/index.js

```diff
@@ -0,0 +1,99 @@
+import fs from 'fs'
+// import path from 'path'
+import LambdaChromeLauncher from './launcher'
+import { debug } from './utils'
+import DEFAULT_CHROME_FLAGS from './flags'
+
+const DEVTOOLS_PORT = 9222
+const DEVTOOLS_HOST = 'http://127.0.0.1'
+
+// Prepend NSS related libraries and binaries to the library path and path respectively on lambda.
+/* if (process.env.AWS_EXECUTION_ENV) {
+  const nssSubPath = fs.readFileSync(path.join(__dirname, 'nss', 'latest'), 'utf8').trim();
+  const nssPath = path.join(__dirname, 'nss', subnssSubPathPath);
+
+  process.env.LD_LIBRARY_PATH = path.join(nssPath, 'lib') +  ':' + process.env.LD_LIBRARY_PATH;
+  process.env.PATH = path.join(nssPath, 'bin') + ':' + process.env.PATH;
+} */
+
+// persist the instance across invocations
+// when the *lambda* container is reused.
+let chromeInstance
+
+export default async function launch ({
+  flags = [], chromePath, port = DEVTOOLS_PORT, forceLambdaLauncher = false,
+} = {}) {
+  const chromeFlags = [...DEFAULT_CHROME_FLAGS, ...flags]
+
+  if (!chromeInstance) {
+    if (process.env.AWS_EXECUTION_ENV || forceLambdaLauncher) {
+      chromeInstance = new LambdaChromeLauncher({
+        chromePath,
+        chromeFlags,
+        port,
+      })
+    } else {
+      // This let's us use chrome-launcher in local development,
+      // but omit it from the lambda function's zip artefact
+      try {
+        // eslint-disable-next-line
+        const { Launcher: LocalChromeLauncher } = require('chrome-launcher')
+        chromeInstance = new LocalChromeLauncher({ chromePath, chromeFlags: flags, port })
```

Comment:
i am doing typescript migration work now, everything okay except this: it makes type mismatch. 
as you know, that LocalChromeLauncher is quiet different with our LambdaLauncher, typescript complains about that because LocalChromeLauncher cannot be casted to our LambdaLauncher type safely.

Here is some example that makes me hard

```typescript
let chromeInstance: LambdaChromeLauncher | undefined;
export default async function launch() {
  // some codes
  if (shouldRunLambdaLauncher) {
    chromeInstance = new LambdaChromeLauncher(...); // okay
  } else {
     const { Launcher: LocalChromeLauncher } = require('chrome-launcher');
     chromeInstance = new LocalChromeLauncher(...); // type mismatch error!
  }
}
```

Reply:
btw specifying chrome-launcher package to devDependencies is one of available solution:

```
import { Launcher: LocalChromeLauncher } from 'chrome-launcher';

let chromeInstance: LambdaChromeLauncher | LocalChromeLauncher | undefined;
export default async function launch() {
  // some codes
  if (shouldRunLambdaLauncher) {
    chromeInstance = new LambdaChromeLauncher(...); // okay
  } else {
     chromeInstance = new LocalChromeLauncher(...); // okay now!
  }
}
```

how about think about this solution?

Reply:
Hi @mooyoul—Hm.. I see.  My concern with importing `chrome-launcher` is that it now has to be added as a `dependency` rather than a `devDependency`? One option is to completely remove the custom `LambdaChromeLauncher` and use `chrome-launcher` for both local-development and Lambda.

Some context: Originally I wanted to use `chrome-launcher` for launching Chrome whether it was locally or on Lambda. However, at the time, the dependency tree of `chrome-launcher` was quite huge, bundling all sorts of Lighthouse stuff which itself had very large dependencies which caused a lot of bloat in the lambda-function-handler's deployment package. Because of this, I created `src/launcher.js` mostly based on the `chrome-launcher` code. However, as was pointed out by @pedro93 in #48, it seems that `chrome-launchers`'s dependency tree is no longer very large (MB-size wise)—so we could revisit using it.

A caveat with switching to using `chrome-launcher` is that, the `LaunchedChrome` interface [does not expose](https://github.com/GoogleChrome/chrome-launcher/blob/master/chrome-launcher.ts#L42) the the reference to the `spawn`ed child process, and there is no option to `unref()`, so you'd have to go behind `chrome-launcher`'s back to `unref()` the spawed `child_process`—code which would likely cause a type error in Typescript..

But.. it's still a trade-off. Right now the `serverless-chrome/lambda` package has very few dependencies—which I quite like, and is important because of the size limitations imposed on us by Lambda. Adding `chrome-launcher` adds quite a number of child dependencies including `rimraf`, which itself has a lot of nested dependencies.. 

What about something as described in the Typescript [Modules' Optional Module Loading and Other Advanced Loading Scenarios documentation](https://www.typescriptlang.org/docs/handbook/modules.html#optional-module-loading-and-other-advanced-loading-scenarios)? (I haven't tried this myself, so no guarantee that it works..)

Reply:
Also, if it's an impossible blocker.. always possible to cheat:

```js
let chromeInstance: any;
```

Yes—not good. But..

Reply:
I needed type definition for chrome-launcher package, because currently chrome-launcher has bundled type definition, not separated package like DefinitelyTyped (e.g. @types/chrome-launcher) so that's the reason why i considered adding chrome-launcher as devDependency.

Reply:
yeah `any` is possible option, even it's not good-looking.